fetch.bs

```diff
@@ -4970,15 +4970,32 @@ steps:
      <a for=url>blob URL entry</a>.
 
      <li>
-      <p>If <var>request</var>'s <a for=request>method</a> is not `<code>GET</code>`,
-      <var>blobURLEntry</var> is null, or <var>blobURLEntry</var>'s
-      <a for="blob URL entry">object</a> is not a {{Blob}} object, then return a
-      <a>network error</a>. [[!FILEAPI]]
+      <p>If <var>request</var>'s <a for=request>method</a> is not `<code>GET</code>` or
+      <var>blobURLEntry</var> is null, then return a <a>network error</a>. [[!FILEAPI]]
 
       <p class=note>The `<code>GET</code>` <a for=/>method</a> restriction serves no useful purpose
       other than being interoperable.
 
-     <li><p>Let <var>blob</var> be <var>blobURLEntry</var>'s <a for="blob URL entry">object</a>.
+     <li><p>Let <var>requestEnvironment</var> be the result of
+     <a for=request>determining the environment</a> given <var>request</var>.
+
+     <li><p>Let <var>isTopLevelNavigation</var> be true if <var>request</var>'s <a
+     for=request>destination</a> is "<code>document</code>"; otherwise, false.
+
+     <li><p>If <var>isTopLevelNavigation</var> is false and <var>requestEnvironment</var> is null,
+     then return a <a>network error</a>.
+
+     <li><p>Let <var>blob</var> be null.
+
+     <li><p>If <var>isTopLevelNavigation</var> is true, then set <var>blob</var> to the result of
+     <a href="https://w3c.github.io/FileAPI/#blob-url-obtain-object">obtaining a blob object</a>
+     given <var>blobURLEntry</var> and the string "<code>navigation</code>".
```

Comment:
I think it would be clearer to first initialize a variable called _navigationOrEnvironment_ and then invoke "obtaining a blob object" once. At least I find code written that way easier to follow.

Reply:
Thanks for the feedback, I wasn't sure if it was OK to assign values of different types into a variable. It's indeed simpler with only the one all to "obtaining a blob object". Thanks, and done!